### PR TITLE
Cloudflare waf integration

### PR DESCRIPTION
**Pull request by TalGumi**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x ] No

## Must have
- [x ] Tests
- [x ] Documentation

### CODE REVIEW

File: Packs/CloudflareWAF/pack_metadata.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "author": "Cortex XSOAR",
+    "url": "https://www.paloaltonetworks.com/cortex",
+    "email": "",
+    "categories": [
+        "Network Security"
+    ],
+    "tags": [],
+    "useCases": [],
+    "keywords": []
+}
```
Comment (Noy-Maimon) [inline Packs/CloudflareWAF/pack_metadata.json:15]: please add the 'marketplaces' field and populate it with the marketplaces this pack is going to be supported by (see example [here](https://github.com/demisto/content/blob/master/Packs/AMP/pack_metadata.json))

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]
Looks great!
Please see my comment about the metadata.

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py [hunk 2]
```diff
@@ -0,0 +1,1212 @@
...
+
+    items = [{'ip': item} for item in argToList(args.get('items'))]
+
+    response = client.cloudflare_waf_ip_list_item_update_request(list_id, items)
+
+    output = response['result']
+
+    return CommandResults(
+        readable_output=f'Replacing items in the IP List {list_id} is executing'
+    ), output
```
Comment (Noy-Maimon) [inline Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:999]: Is it possible that a few lines here were deleted unintentionally?

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py (hunk 2 above)
Comment (TalGumi) [reply Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:999]: It is deleted after meeting with you, at your request.

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py [hunk 3]
```diff
@@ -0,0 +1,1212 @@
...
```
Comment (Noy-Maimon) [inline Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:999]:
```suggestion
    return CommandResults(
        readable_output=f'Replace items in the IP List {list_id} is executing',
        raw_response=output)
   
```

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py [hunk 4]
```diff
@@ -0,0 +1,1212 @@
...
+    list_id = args['list_id']
+    items = [{'id': item} for item in argToList(args.get('items_id'))]
+
+    response = client.cloudflare_waf_ip_list_item_delete_request(list_id, items)
+
+    output = response['result']
+
+    return CommandResults(
+        readable_output=f'Deleting items from ip-list {list_id} is executing'
+    ), output
```
Comment (Noy-Maimon) [inline Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:1022]:
```suggestion
    return CommandResults(
        readable_output=f'Delete items from ip-list {list_id} is executing',
        raw_response=output)
        
```

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py [hunk 5]
```diff
@@ -0,0 +1,1212 @@
...
+
+    list_id = args['list_id']
+    items = [{'ip': item} for item in argToList(args.get('items'))]
+
+    response = client.cloudflare_waf_ip_list_item_create_request(list_id, items)
+    output = response['result']
+
+    return CommandResults(
+        readable_output=f'Adding items to the ip-list {list_id} is executing'
+    ), output
```
Comment (Noy-Maimon) [inline Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:975]:
```suggestion
    return CommandResults(
        readable_output=f'Add items to the ip-list {list_id} is executing',
        raw_response=output)
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]
Hi, please see these small changes i added

**Review by Noy-Maimon — APPROVED** [review]

**Pull request merged**